source/Pages/compute_on_spider.rst

```diff
@@ -305,7 +318,408 @@ combination that is not available you will receive the following error message:
+
+.. WARNING::
+   As the home folder is mounted by default in singularity, and python searches certain folders by default, it is possible that inside the container packages from the host machine are called, instead of what is inside the container. For example, the ``~/.local`` folder on the host machine can have presedence over site-packages in the container. If errors appear relating to CUDA ``.so`` files, or versions of packages are mismatching, ensure that the user-space is not accidentally providing libraries to the container.
+
+.. tip::
+   Use singularity only to control the versioning of the environment and encapsulate your libraries in the container and thus control their versioning. Code and data files can be fed to singularity, so keep such files external to the container.
+
+
+The example we are about to execute in the container comes from the tensorflow library: `classifying pieces of clothing <https://www.tensorflow.org/tutorials/keras/classification>`_. Now create a file to run ``fashion.py``, set it to executable with ``chmod 755 fashion.py`` and add the following:
+
+.. code-block:: python
+
+  #!/usr/bin/env python
+
+  # TensorFlow and tf.keras
+  import tensorflow as tf
+
+  # Helper libraries
+  import numpy as np
+  import matplotlib.pyplot as plt
+
+  print(tf.__version__)
+
+  fashion_mnist = tf.keras.datasets.fashion_mnist
+
+  (train_images, train_labels), (test_images, test_labels) = fashion_mnist.load_data()
+
+  class_names = ['T-shirt/top', 'Trouser', 'Pullover', 'Dress', 'Coat',
+                 'Sandal', 'Shirt', 'Sneaker', 'Bag', 'Ankle boot']
+
+  train_images = train_images / 255.0
+  test_images = test_images / 255.0
+
+  model = tf.keras.Sequential([
+      tf.keras.layers.Flatten(input_shape=(28, 28)),
+      tf.keras.layers.Dense(128, activation='relu'),
+      tf.keras.layers.Dense(10)
+  ])
+
+  model.compile(optimizer='adam',
+                loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+                metrics=['accuracy'])
+
+  model.fit(train_images, train_labels, epochs=10)
+
+  test_loss, test_acc = model.evaluate(test_images,  test_labels, verbose=2)
+  print('\nTest accuracy:', test_acc)
+
+  probability_model = tf.keras.Sequential([model,
+                                           tf.keras.layers.Softmax()])
+
+  predictions = probability_model.predict(test_images)
+  print(predictions[0])
+
+This example will create a model that recognizes the clothes in a picture, and a prediction of a set of test images is done at the end. The result can be compared to the `official example <https://www.tensorflow.org/tutorials/keras/classification>`_. The matplotlib output is omitted in this example for simplicity. This output can be seen in the section on :ref:`jupyter notebooks <jupyter-notebooks>`.
+
+Now this code can be run with:
+
+.. code-block:: bash
+
+   singularity exec --nv nv-tf-22.07.sif ./fashion.py
```

Comment:
perhaps make it clear that this ofcourse should be run on the GPU node via sbatch or via srun